models/message.go

```diff
@@ -0,0 +1,51 @@
+package models
+
+import (
+	"bytes"
+	"encoding/base32"
+
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
+)
+
+type NewMessageResponse struct {
+	Id      string `json:"id"`
+	Message string `json:"message"`
+}
+
+type Message struct {
+	Id      string `json:"id"`
+	Subject string `json:"subject"`
+	Text    string `json:"text"`
+	From    string `json:"from"`
+	To      string `json:"recipients"`
+	Status  string `json:"status"`
+}
+
+type QueueMessage struct {
+	Id   string `json:"id"`
+	Type string `json:"type"`
+}
+
+// After marshaling from JSON, call this method to validate the object is intact
+func (self *Message) Validate() error {
+
+	if err := ValidEmail(self.From); err != nil {
+		return errors.Wrap(err, "From")
+	}
+
+	if err := ValidEmail(self.To); err != nil {
+		return errors.Wrap(err, "To")
+	}
+
+	return nil
+}
+
+func NewId() string {
```

Comment:
My goal for the ID was to generate a random alpha-numeric value the user can easily copy and paste. I could have just used a uuid for this purpose, but I wanted to show I had some experience with base encoding.

After looking at the "hash/fnv" package I'm not sure how I would use that to generate a random alpha-numeric string. It's designed for use in hashing existing values. 

However while looking at this I did discover ```rand.Read()``` which would remove my dependency on ```github.com/pborman/uuid``` which I like.

Code would look like this
```go
package main

import (
	"bytes"
	"crypto/rand"
	"encoding/base32"
	"fmt"
)

func main() {
	var buf bytes.Buffer

	randBuf := make([]byte, 26)
	rand.Read(randBuf)
	encoder := base32.NewEncoder(base32.StdEncoding, &buf)
	encoder.Write(randBuf)
	encoder.Close()
	buf.Truncate(26)

	fmt.Printf("ID: %s\n", buf.String())
}
```
Output: 
```
$ go build -o hash hash.go && ./hash
ID: EM7KA7DVRHXDY4W5AWCPZR66HQ
```